src/debuglog.h

```diff
@@ -0,0 +1,52 @@
+/*
+    Copyright (C) 2020 Sebastian J. Wolf and other contributors
+
+    This file is part of Fernschreiber.
+
+    Fernschreiber is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    Fernschreiber is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with Fernschreiber. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef DEBUGLOG_H
+#define DEBUGLOG_H
+
+
+#include <QObject>
+#include <QQmlEngine>
+
+class DebugLog : public QObject
+{
+    Q_OBJECT
+    Q_PROPERTY(bool enabled READ isEnabled WRITE setEnabled NOTIFY enabledChanged)
+public:
+
+#ifdef QT_QML_DEBUG
+    DebugLog(QObject* parent = Q_NULLPTR) : QObject(parent), enabled(true) {}
+#else
+    DebugLog(QObject* parent = Q_NULLPTR) : QObject(parent), enabled(false) {}
```

Comment:
How about initializing `enabled` from an environment variable?

Reply:
Yeah, I wasn't sure how you wanted to do the "real" c++ part and left that one as basic as possible as a result.

I thought we could use `QCommandLineParser` to introduce a `-v` argument for non-debug builds, possibly in addition to environment variables? What do you suggest?

Sorry for the noob question, but I guess, the `static` declaration would be enough to ensure it's the same instance if I called `createSingleton` from `harbour-fernschreiber.cpp` again, right?

Reply:
`DebugLog::createSingleton` (or whatever is passed to `qmlRegisterSingletonType<DebugLog>`) is called by QML engine when it needs to access `DebugLog` object for the first time. Lazy instantiation on demand. QML engine owns this object and will destroy it at shutdown. There's no guarantee when (if ever) this singleton will be instantiated. Which makes it a bit inconvenient to interact with this instance from C++ side.

The `enabled` field can be initialized from a global variable (which in turn can be initialized in `main` before even registering the singleton). Alternatively or in addition to that, `DebugLog::createSingleton` can set a global variable pointing to this singleton instance instantiated and owned by QML engine, the destructor can set it to NULL (checking first that it's equal to `this`, just in case) if it needs to be accessed from C++ later in the game (which is probably unnecessary). Something along these lines can be done.

Reply:
Or you can just leave it as is (leaving it enabled by default) and I can integrate that later with logging categories.

Reply:
Yeah, maybe that would get us a better result ;)